packages/ra-ui-materialui/src/field/BooleanField.tsx

```diff
@@ -27,15 +27,16 @@ const useStyles = makeStyles(
 
 const BooleanField = (props: BooleanFieldProps) => {
     const {
+        addLabel = true,
```

Comment:
I believe this needs to be set on the exported component (the memo)

Reply:
Any hint on how to do this since React.memo does not support defaultProps, only the inner component does.

Reply:
This was how I landed on moving the default logic to the component that clones the component because there didn't appear to be a good way of doing this whilst maintaining the memo.